From 9b12305f591cd1f3fb03521e6d21dd5ec9c8fb6e Mon Sep 17 00:00:00 2001
From: Dong Aisheng <aisheng.dong@nxp.com>
Date: Fri, 24 Nov 2017 16:51:50 +0800
Subject: [PATCH 2959/5242] MLK-17074-4 soc: imx8: pm-domains: remove the
 status checking during power off

commit  6fa8c018343a462c1af233188c7cf4c52576905a from
https://source.codeaurora.org/external/imx/linux-imx.git

The power domain core alreadys checked it, no need check it anymore.
Besides that, removing it make the driver be able to switch to different
low power mode in the future. Identically the power on check is also
removed.

Reviewed-by: Frank Li <frank.li@nxp.com>
Reviewed-by: Ranjani Vaidyanathan <Ranjani.Vaidyanathan@nxp.com>
Signed-off-by: Dong Aisheng <aisheng.dong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/imx/pm-domains.c |    6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/soc/imx/pm-domains.c b/drivers/soc/imx/pm-domains.c
index ed8a1b2..b87272f 100644
--- a/drivers/soc/imx/pm-domains.c
+++ b/drivers/soc/imx/pm-domains.c
@@ -86,8 +86,7 @@ static int imx8_pd_power_on(struct generic_pm_domain *domain)
 
 	ret = imx8_pd_power(domain, true);
 
-	if (!list_empty(&pd->clks) && domain->status == GPD_STATE_POWER_OFF
-		&& !pd->runtime_idle_active) {
+	if (!list_empty(&pd->clks) && !pd->runtime_idle_active) {
 		/*
 		 * The SS is powered on restore the clock rates that
 		 * may be lost.
@@ -122,8 +121,7 @@ static int imx8_pd_power_off(struct generic_pm_domain *domain)
 
 	pd = container_of(domain, struct imx8_pm_domain, pd);
 
-	if (!list_empty(&pd->clks) && (domain->status != GPD_STATE_POWER_OFF)
-		&& (!pd->runtime_idle_active)) {
+	if (!list_empty(&pd->clks) && (!pd->runtime_idle_active)) {
 		/*
 		 * The SS is going to be powered off, store the clock rates
 		 * that may be lost.
-- 
1.7.9.5

